ocfweb/docs/views/servers.py

```diff
@@ -57,81 +56,74 @@ def has_munin(self):
 def get_hosts():
-    return [
-        Host.from_ldap(
-            hostname='hal',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'fallingrocks',
-                    'monsoon',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='jaws',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'anthrax',
-                    'biohazard',
-                    'death',
-                    'dementors',
-                    'democracy',
-                    'firestorm',
-                    'flood',
-                    'lightning',
-                    'maelstrom',
-                    'pestilence',
-                    'reaper',
-                    'supernova',
-                    'thunder',
-                    'tsunami',
-                    'werewolves',
-                    'whirlwind',
-                    'whiteout',
-                    'zombies',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='pandemic',
-            type='hypervisor',
-            children=[
-                Host.from_ldap(hostname)
-                for hostname in (
-                    'pileup',
-                )
-            ],
-        ),
-
-        Host.from_ldap(
-            hostname='riptide',
-            type='hypervisor',
-        ),
-
-        Host.from_ldap(
-            hostname='corruption',
-            type='server',
-        ),
-
-        Host('blackhole', 'network', 'Managed Cisco Catalyst 2960S-48TS-L Switch.', []),
-
-        Host('logjam', 'printer', '', []),
-        Host('pagefault', 'printer', '', []),
-        Host('papercut', 'printer', '', []),
-        Host.from_ldap('overheat', type='raspi'),
-        Host.from_ldap('tornado', type='nuc'),
-
-        *(Host.from_ldap(desktop, type='desktop')
-          for desktop in sorted(list_desktops())),
-    ]
+    def is_hidden(host):
+        return host['cn'][0][:6] == 'hozer-' or host['cn'][0][:4] == 'dev-'
+
+    def create_hosts(lst):
+        """Accepts a list of raw ldap output, returns a dictionary of host
+        objects indexed by hostname
+        """
+        hosts = {}
+        for h in lst:
+            if not is_hidden(h):
+                description = h.get('description', [''])[0]
+                hostname = h['cn'][0]
+                hosts[hostname] = (Host(hostname, h['type'], description, ()))
+        return hosts
+
+    servers = create_hosts(hosts_by_filter('(type=server)'))
+    desktops = create_hosts(hosts_by_filter('(type=desktop)'))
+    misc = create_hosts(hosts_by_filter('(type=printer)'))
+    hypervisors = {}
+
+    # Add children to hypervisors
+    for h in list(servers.values()):
+        children = []
+        for child_hostname in get_children(h.hostname):
```

Comment:
This will cause you to query PuppetDB once for every server in the servers list. That's a lot of queries - slow!

You could fix this by implementing some sort of caching for `get_children`. But an even better way would be to rearchitect this code so it queries PuppetDB only once, gets all the information it needs, and then processes the response to extract the needed information.